dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/generate-keying-data/generate-keying-data.event-handler.test.ts

```diff
@@ -37,50 +36,168 @@ describe('handleUtilisationReportGenerateKeyingDataEvent', () => {
     jest.resetAllMocks();
   });
 
-  it('updates each fee record in the payload generateKeyingDataDetails', async () => {
+  it('updates each fee record in the payload and calculates the keying data columns when all the supplied fee records have a unique facility id', async () => {
     // Arrange
     const utilisationReport = aReconciliationInProgressReport();
-    const feeRecords = aListOfFeeRecordsForReport(utilisationReport);
+    const feeRecords = [
+      aMatchedFeeRecordForReportWithFacilityId(utilisationReport, '11111111'),
+      aMatchedFeeRecordForReportWithFacilityId(utilisationReport, '22222222'),
+      aMatchedFeeRecordForReportWithFacilityId(utilisationReport, '33333333'),
+    ];
     const eventHandlers = feeRecords.reduce((obj, { id }) => ({ ...obj, [id]: aMockEventHandler() }), {} as { [id: number]: jest.Mock });
     const feeRecordStateMachines = feeRecords.reduce(
       (stateMachines, { id }) => ({ ...stateMachines, [id]: aMockFeeRecordStateMachine(eventHandlers[id]) }),
       {} as { [id: number]: FeeRecordStateMachine },
     );
 
-    const generateKeyingDataDetails: GenerateKeyingDataDetails = feeRecords.map((feeRecord) => ({ feeRecord, generateKeyingData: false }));
-
     jest.spyOn(FeeRecordStateMachine, 'forFeeRecord').mockImplementation((feeRecord) => feeRecordStateMachines[feeRecord.id]);
+    mockFind.mockResolvedValue([]);
 
     // Act
     await handleUtilisationReportGenerateKeyingDataEvent(utilisationReport, {
       transactionEntityManager: mockEntityManager,
-      generateKeyingDataDetails,
+      matchFeeRecords: feeRecords,
       requestSource,
     });
 
     // Assert
-    feeRecords.forEach(({ id }, index) => {
+    expect(mockFind).toHaveBeenCalledWith(FeeRecordEntity, {
+      where: {
+        report: { id: utilisationReport.id },
+        status: In(['TO_DO', 'DOES_NOT_MATCH']),
+      },
+    });
+
+    feeRecords.forEach(({ id }) => {
       const eventHandler = eventHandlers[id];
-      const { generateKeyingData } = generateKeyingDataDetails[index];
       expect(eventHandler).toHaveBeenCalledWith({
-        type: 'KEYING_DATA_GENERATED',
+        type: 'GENERATE_KEYING_DATA',
+        payload: {
+          transactionEntityManager: mockEntityManager,
+          isFinalFeeRecordForFacility: true,
+          requestSource,
+        },
+      });
+    });
+  });
+
+  describe('when there are multiple fee records with the same facility id', () => {
+    const facilityId = '12345678';
+
+    it('calculates keying data only for the first fee record in the payload', async () => {
```

Comment:
This might be a tad pedantic, but there is no requirement for it to be the first, that is an implementation choice, so (leaving the body of the test the same) could we rename this to 'calculates keying data for only one of the fee records in the payload with that facility id' please?